cea/optimization_new/energyPotential.py

```diff
@@ -0,0 +1,193 @@
+            return np.nan
+        # initialise necessary variables
+        nbr_of_files = len(energy_potential_files)
+        average_temps = [np.nan] * nbr_of_files
+        self.main_potential = pd.DataFrame(0.0, index=np.arange(self.time_frame),
+                                           columns=pd.concat([pd.Series(['domain_potential']), building_codes]))
+        if auxiliary_potential_column_name is not None:
+            self.auxiliary_potential = pd.DataFrame(0.0, index=np.arange(self.time_frame),
+                                                    columns=pd.concat([pd.Series(['domain_potential']), building_codes]))
+        # if specific potential file for a building exists, save potential to object attribute (pd.Dataframe)
+        for (file, i) in zip(energy_potential_files, np.arange(nbr_of_files)):
+            if exists(file):
+                pvt_potential = pd.read_csv(file)
+                self.main_potential[building_codes[i]] = pvt_potential[main_potential_column_name]
+                if temperature_column_name is not None:
+                    average_temps[i] = self._get_average_temp(pvt_potential[temperature_column_name], building_codes[i])
+                if auxiliary_potential_column_name is not None:
+                    self.auxiliary_potential[building_codes[i]] = pvt_potential[auxiliary_potential_column_name]
+        # calculate total potential across the domain
+        self.main_potential['domain_potential'] = self.main_potential[building_codes].sum(axis=1)
+        if auxiliary_potential_column_name is not None:
+            self.auxiliary_potential['domain_potential'] = self.auxiliary_potential[building_codes].sum(axis=1)
+        # if thermal(!) energy potential: return average return temperature
+        if temperature_column_name is not None:
+            average_temperature = np.nanmean(average_temps)
+            return average_temperature
+
+    def _get_average_temp(self, temperature_series, building_code=None):
+        average_temp = np.mean(temperature_series)
+        if all(np.isnan(temperature_series)):
+            if building_code is None:
+                print(f"There seems to be a problem with the {self.type} potential! "
+                      f"Please check the corresponding .csv-file.")
+            else:
+                print(f"There seems to be a problem with the {self.type} potential of building {building_code}! "
+                      f"Please check the corresponding .csv-file.")
+        elif average_temp == 0:
+            average_temp = np.nan
+        return average_temp
+
+    @staticmethod
+    def _temp_to_thermal_ec(temperature, thermal_energy_carriers):
+        if not np.isnan(temperature):
+            index_closest_mean_temp = (thermal_energy_carriers['mean_qual'] - temperature).abs().nsmallest(n=1).index[0]
+            energy_carrier_code = thermal_energy_carriers['code'].iloc[index_closest_mean_temp]
+        else:
+            energy_carrier_code = "T???"
```

Comment:
placeholder?

Reply:
Yes, that is a placeholder for now. I'm currently linking that up. Should be part of the next bigger commit at the end of this week.